stable/datadog/templates/agent-clusterchecks-clusterrolebinding-auth-delegator.yaml

```diff
@@ -0,0 +1,19 @@
+{{- if and .Values.rbac.create .Values.clusterAgent.enabled .Values.clusterAgent.clusterChecks.enabled  .Values.clusterAgent.metricsProvider.enabled .Values.clusterchecksDeployment.enabled -}}
```

Comment:
Why checking `Values.clusterAgent.metricsProvider.enabled` ?

Reply:
can be removed I think

Reply:
In fact after looking again in the current RBAC file, the condition need to check 'Values.clusterAgent.metricsProvider.enabled' to be align the condition present in the equivalent file: https://github.com/helm/charts/blob/master/stable/datadog/templates/agent-clusterrolebinding-auth-delegator.yaml#L1

Reply:
in fact it is a useless permission for the agent running clusterchecks. I will remove file this file.
the other `...auth-delegator.yaml` file is for the `cluster-agent` and not the agent